### ISSUE TITLE
Archives fail to open in osx Archive Utility on versions >=0.0.52

### ISSUE THREAD

**Issue by scott-force** [uktrade/stream-zip#27]
I've been developing a feature with stream-zip and everything had been working great for about a month. I noticed after a recent library update that attempting to open the archive created by stream-zip failed to open on osx with `Error 79 - Inappropriate file type or format`. If i use the `unzip` cmd in the terminal, it unzips fine. IntelliJ is also able to peek into the archive and show me the data inside, so it appears things are still "mostly" working. 

If I roll back to library 0.0.51 or below, the error goes away completely.

I'm running on an M1 Macbook Pro on Ventura 13.2.1, but others have reproduced the issue on slightly differing versions of the OS.

steps to reproduce:
slightly modifying the example on the documentation page can trigger this error. I tested with ZIP_32 and ZIP_64 compression:

```py
from datetime import datetime
from stream_zip import ZIP_32, stream_zip

def member_files():
    modified_at = datetime.now()
    perms = 0o600

    def file_1_data():
        yield b'Some bytes 1'

    yield ('my-file-1.txt', modified_at, perms, ZIP_32, file_1_data())

zipped_chunks = stream_zip(member_files()):

with open('test.zip', 'wb') as fp:
    for zipped_chunk in zipped_chunks:
        fp.write(zipped_chunk)
```

additional request: I can't remember if a changelog was present prior to version 54, but it's hard to track down any potential reasons for this change without a broader context on major changes 😅 
is it possible to bring back notes for versions < 54?

**Comment by michalc** [comment]
Hi @scott-force,

Thanks for the report. Firstly

> is it possible to bring back notes for versions < 54?

So far we've never had a changelog, but more depending on the commit messages. We should probably have a more visible changelog though yes.

But on the main issue:

The main (or maybe even only) feature/code change after 51 was this one:

https://github.com/uktrade/stream-zip/commit/1130fdcb5f8a068c194392fdf5b0d8b6769a8928

Although interestingly it _should_ have only affected ZIP_64. Just to absolutely confirm you get the issue on both ZIP_32 and ZIP_64?

(I will have a nose around and see if I can reproduce the issue myself in the meantime)

Thanks,

Michal

**Comment by michalc** [comment]
> Although interestingly it should have only affected ZIP_64. Just to absolutely confirm you get the issue on both ZIP_32 and ZIP_64?
>
> (I will have a nose around and see if I can reproduce the issue myself in the meantime)

I can reproduce the issue using ZIP_64, but with ZIP_32 as in your example, it seems to work fine

**Comment by michalc** [comment]
I'm tempted to basically revert https://github.com/uktrade/stream-zip/commit/1130fdcb5f8a068c194392fdf5b0d8b6769a8928, but I would love a test on it somehow...

**Comment by scott-force** [comment]
Thanks for the fast reply - you are correct, I made a mistake in my testing and accidentally used the ZIP_64 data twice. ZIP_32 does in fact work well in my environment.

I am happy to help test in any way I can - I can manually make the change in https://github.com/uktrade/stream-zip/commit/1130fdcb5f8a068c194392fdf5b0d8b6769a8928 and see how it goes if that helps

**Comment by michalc** [comment]
I've made the change in https://github.com/uktrade/stream-zip/pull/28. If you're happy to give that a test, that would be helpful?

**Comment by michalc** [comment]
(Mostly just thinking out loud)

Aha - I've found that the cpio from libarchive also doesn't like the zip file made from your example with ZIP_64, and  stream-zip v0.0.56, but works fine with the change in https://github.com/uktrade/stream-zip/pull/28

Specifically

```
cpio -i < test.zip
```

gives

```
cpio: Inconsistent uncompressed size: 12 in central directory, 4294967295 in local header: Inappropriate file type or format
```

I can probably come up with a test

(Also it makes me wonder if Mac's Archive Utility uses libarchive under the hood...)

**Comment by scott-force** [comment]
As expected I was able to successfully extract an archive built with ZIP_64 using the branch from https://github.com/uktrade/stream-zip/pull/28

Sounds like you've figured out how to prevent regression going forward - that's great 😄  thank you for the very quick resolution

**Comment by michalc** [comment]
Released fix in v0.0.57

- https://pypi.org/project/stream-zip/0.0.57/
- https://github.com/uktrade/stream-zip/releases/tag/v0.0.57
